presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/Util.java

```diff
@@ -87,4 +88,54 @@ public static PlanNode transpose(PlanNode parent, PlanNode child)
                         node,
                         Assignments.identity(restrictedOutputs)));
     }
+
+    /**
+     * @return The original node, with identity projections possibly inserted between node and the child, limiting the columns to those permitted.
+     * Returns a present Optional iff the child was rewritten.
+     */
+    public static Optional<PlanNode> restrictChildOutputs(PlanNodeIdAllocator idAllocator, PlanNode node, Set<Symbol> permittedChildOutputs)
+    {
+        return restrictChildrenOutputs(idAllocator, node, ImmutableList.of(permittedChildOutputs));
+    }
+
+    /**
+     * @return The original node, with identity projections possibly inserted between node and each child, limiting the columns to those permitted.
+     * Returns a present Optional iff at least one child was rewritten.
+     */
+    public static Optional<PlanNode> restrictBinaryChildrenOutputs(
```

Comment:
do you need emphases with `Binary` in name that it supports binary children nodes? IMO the fact it accepts two parameters is enough. 

Well I would go even one step further and create a method which accepts a var-args parameter or just a list, and then check if number of children matches to the number of parameters.